### PR TITLE
Switch snafu to thiserror

### PR DESCRIPTION
**Pull request by LovecraftianHorror**
This is an experimental switch to `thiserror` `from` `snafu` (following #71). The good news is this drops compile times, from `cargo bloat -j 1 --time --release` it went from a total of 156.58 to 93.16 which is a pretty significant drop (on my `Intel i5-5200U @ 2.7GHz` CPU). I would double check everything to make sure anything is different.

### CODE REVIEW

**Comment by LovecraftianHorror** [comment]
CI failure is from an unrelated `clippy` lint

File: src/process/errors.rs [hunk 1]
```diff
@@ -1,24 +1,22 @@
...
 
-use snafu::Snafu;
-
 use crate::{Error, Pid};
 
 pub type ProcessResult<T> = std::result::Result<T, ProcessError>;
 
-#[derive(Debug, Snafu)]
-#[snafu(visibility(pub(crate)))]
+// TODO: get this visibility junk sorted out
```
Comment (cjbassi) [inline src/process/errors.rs:7]: Is this still an issue? I didn't think thiserror had to deal with any visibility shenanigans.

Comment (LovecraftianHorror) [reply src/process/errors.rs:7]: Ah yeah, sorry I forgot to clean up that comment before I submitted. I can remove tomorrow.

**Comment by cjbassi** [comment]
This looks really great. I think thiserror is definitely an improvement over snafu esp with the compile time reductions which are quite substantial and the refactoring you did looks better too.

I still have a bit more to review, and I would also like to merge #79 before this PR since there's some additional error handling in that PR that we're going to need to port over.

**Comment by cjbassi** [comment]
Thanks!

File: src/process/errors.rs (hunk 1 above)
Comment (cjbassi) [reply src/process/errors.rs]: Just pushed a commit to fix this.

**Pull request merged**